samples/manage/azure-automation-automated-export/AutoExport.ps1

```diff
@@ -0,0 +1,280 @@
+	$currentTime = Get-Date -format "_yyyy-MM-dd_HH:mm.ss";
+	$dbCopyName = $databaseName + $currentTime;
+    # Add the DatabaseCopyName property and set it.
+	$dbObj | Add-Member -type NoteProperty -name DatabaseCopyName -Value $dbCopyName;
+    # Add the ServerName property and set it.
+    $dbObj | Add-Member -type NoteProperty -name ServerName -Value $serverName;
+    # Add the Export property and set it to $null for now. This will be used to look up the export after it has been started.
+    $dbObj | Add-Member -type NoteProperty -name Export -Value $null;
+    # Add the DatabaseState property and set it to ToCopy so that the "state machine" knows to start the copy of the database.
+	$dbObj | Add-Member -type NoteProperty -name DatabaseState -Value ([DatabaseState]::ToCopy);
+    # Add the RetryCount property and set it to 0. This will be used to count the number of time we retry each failable operation.
+    $dbObj | Add-Member -type NoteProperty -name RetryCount -Value 0;
+    # Add the OperationStartTime property and set it to $null for now. This will be used when an operation starts to correcly do timeouts.
+    $dbObj | Add-Member -type NoteProperty -name OperationStartTime -Value $null;
+
+    # Return the newly created object.
+    return $dbObj;
+}
+
+# This function starts the copy of the database. If there is an error, we set the state to ToDrop. Otherwise, we set the state to Copying.
+function StartCopy($dbObj)
+{
+    # Start the copy of the database.
+	Start-AzureSqlDatabaseCopy -ServerName $dbObj.ServerName -DatabaseName $dbObj.DatabaseName -PartnerDatabase $dbObj.DatabaseCopyName;
+    # $? is true if the last command succeeded and false if the last command failed. If it is false, go to the ToDrop state.
+	if(-not $? -and $global:retryLimit -ile $dbObj.RetryCount)
+	{
+		echo ("Error occurred while starting copy of " + $dbObj.DatabaseName + ". It will not be copied. Deleting the database copy named " + $dbObj.DatabaseCopyName + ".");
+		# Set state to ToDrop in case something does get copied.
+		$dbsCopying[$i].DatabaseState = ([DatabaseState]::ToDrop);
+        # Return so we don't execute the rest of the function.
+		return;
+	}
+    elseif(-not $?)
+    {
+        # We failed but we haven't hit the retry limit yet so increment RetryCount and return so we try again.
+        echo ("Retrying with database " + $dbObj.DatabaseName);
+        $dbObj.RetryCount++;
+        return;
+    }
+	# Set the state of the database object to Copying.
+	$dbObj.DatabaseState = ([DatabaseState]::Copying);
+	echo ("Copying " + $dbObj.DatabaseName);
+    $dbObj.OperationStartTime = Get-Date;
+}
+
+# This function checks the progress of the copy. If there is an error, we set the state to ToDrop. Otherwise, we set the state to ToExport.
+function CheckCopy($dbObj)
+{
+	# Get the status of the database copy.
+	$check = Get-AzureSqlDatabaseCopy -ServerName $dbObj.ServerName -DatabaseName $dbObj.DatabaseName;
```

Comment:
Not likely, but it is possible that there is other DB copy going on that is not started by this script. So it is better qualified with the partner DB as well.